sklearn/model_selection/tests/test_validation.py

```diff
@@ -247,6 +252,106 @@ def test_cross_val_score():
+                        scoring=(make_scorer(precision_score),))
+
+    # So should empty lists/tuples
+    assert_raises_regex(ValueError, error_message_regexp,
+                        cross_val_score, estimator, X, y, scoring=())
+
+    # So should duplicated entries
+    assert_raises_regex(ValueError, error_message_regexp,
+                        cross_val_score, estimator, X, y,
+                        scoring=('f1_micro', 'f1_micro'))
+
+    error_message_regexp = (".*should either be.*string or callable.*for "
+                            "single.*.*dict.*for multi.*")
+
+    # Empty dict should raise invalid scoring error
+    assert_raises_regex(ValueError, error_message_regexp,
+                        cross_val_score, estimator, X, y, scoring=(dict()))
+
+    # And so should any other invalid entry
+    assert_raises_regex(ValueError, error_message_regexp,
+                        cross_val_score, estimator, X, y, scoring=5)
+
+    multiclass_scorer = make_scorer(precision_recall_fscore_support)
+
+    # Multiclass Scorers that return multiple values are not supported yet
+    assert_raises_regex(ValueError,
+                        "Can't handle mix of binary and continuous",
+                        cross_val_score, estimator, X, y,
+                        scoring=multiclass_scorer)
+    assert_raises_regex(ValueError,
+                        "Can't handle mix of binary and continuous",
+                        cross_val_score, estimator, X, y,
+                        scoring={"foo": multiclass_scorer})
+
+    multivalued_scorer = make_scorer(confusion_matrix)
+
+    # Multiclass Scorers that return multiple values are not supported yet
+    assert_raises_regex(ValueError,
+                        "scoring must return a number, got",
+                        cross_val_score, SVC(), X, y,
+                        scoring=multivalued_scorer)
+    assert_raises_regex(ValueError,
+                        "scoring must return a number, got",
+                        cross_val_score, SVC(), X, y,
+                        scoring={"foo": multivalued_scorer})
+
+
+def test_cross_val_score_multiple_metric():
+    # Regression
+    X, y = make_regression(n_samples=30, random_state=0)
+    reg = Ridge(random_state=0)
+
+    # List scoring
+    all_scoring = (('r2', 'neg_mean_squared_error'),
+                   {'r2': make_scorer(r2_score),
+                    'neg_mean_squared_error': 'neg_mean_squared_error'})
+
+    r2_scores = cross_val_score(reg, X, y, cv=5, scoring='r2')
+    mse_scores = cross_val_score(reg, X, y, cv=5,
+                                 scoring='neg_mean_squared_error')
+
+    for scoring in all_scoring:
+        multi_scores = cross_val_score(reg, X, y, cv=5, scoring=scoring)
+        assert_array_almost_equal(multi_scores['r2'], r2_scores)
+        assert_array_almost_equal(multi_scores['neg_mean_squared_error'],
+                                  mse_scores)
+
+    # Classification
```

Comment:
Shrug. I'm not convinced we need to test regression separate from classification.